### PR TITLE
feat(cts): support CTS key event notification resource

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
support new resource **huaweicloud_cts_notification**

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/cts' TESTARGS='-run TestAccCTSNotification_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cts -v -run TestAccCTSNotification_basic -timeout 360m -parallel 4
=== RUN   TestAccCTSNotification_basic
=== PAUSE TestAccCTSNotification_basic
=== CONT  TestAccCTSNotification_basic
--- PASS: TestAccCTSNotification_basic (80.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cts       80.757s
```

### CODE REVIEW

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile("^[\u4e00-\u9fa50-9a-zA-Z_]+$"),
```
Comment (hmbig2) [inline huaweicloud/services/cts/resource_huaweicloud_cts_notification.go:50]: It is different with doc.  `Chinese characters `

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+	if response.Notifications == nil || len(*response.Notifications) == 0 {
+		d.SetId("")
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary:  "Resource not found",
+				Detail:   fmt.Sprintf("cannot retrieve CTS key events notification %s", notificationName),
+			},
+		}
```
Comment (hmbig2) [inline huaweicloud/services/cts/resource_huaweicloud_cts_notification.go]: s/return diag.Diagnostics{/ common.CheckDeletedDiag(d, **new404Error**, "error retrieving CTS key events notification")

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+
+	if ctsNotification.NotificationName == nil {
+		return diag.Errorf("error retrieving CTS key events notification: id not found in api response")
+	}
+
+	d.Set("name", ctsNotification.NotificationName)
+	d.Set("region", region)
+	d.Set("name", ctsNotification.NotificationName)
+	d.Set("notification_id", ctsNotification.NotificationId)
+	d.Set("smn_topic", ctsNotification.TopicId)
```
Comment (hmbig2) [inline huaweicloud/services/cts/resource_huaweicloud_cts_notification.go]: pls use multierror.Append

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+			"notification_users": {
```
Comment (hmbig2) [inline huaweicloud/services/cts/resource_huaweicloud_cts_notification.go:85]: this is a

File: docs/resources/cts_notification.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  **customized**.
+
+* `smn_topic` - (Optional, String) Specifies the URN of a topic.
+
+* `operations` - (Optional, List) Specifies an array of operations that will trigger notifications.
+  For details, see [Supported Services and Operations](https://support.huaweicloud.com/intl/en-us/usermanual-cts/cts_03_0022.html).
+  The [object](#notification_operations_object) structure is documented below.
+
+* `notification_users` - (Optional, List) Specifies an array of users. SMN will notify subscribers of key operations
+  initiated by the specified users. The [object](#notification_users_object) structure is documented below.
```
Comment (hmbig2) [inline docs/resources/cts_notification.md]: pls add default: trace all users.   s/notification_users/trace_users

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/cts/resource_huaweicloud_cts_notification.go]: fixed

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go (hunk 3 above)
Comment (ShiChangkuo) [reply huaweicloud/services/cts/resource_huaweicloud_cts_notification.go]: ok

File: huaweicloud/services/cts/resource_huaweicloud_cts_notification.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/cts/resource_huaweicloud_cts_notification.go:50]: fixed

File: docs/resources/cts_notification.md (hunk 5 above)
Comment (ShiChangkuo) [reply docs/resources/cts_notification.md]: rename it to **operation_users**

**Pull request merged**